### PR TITLE
build: Netty 4.1.87.Final

### LINKED ISSUE #8773: Upgrade to Netty 4.1.90.Final
### Feature description

New netty patch release out

### PR DESCRIPTION
**Pull request by sdelamo**
Closes#8773

Releases:

- [Netty 4.1.89](https://netty.io/news/2023/02/13/4-1-89-Final.html) 
- [Netty 4.1.88](https://netty.io/news/2023/02/12/4-1-88-Final.html) 
- [Netty 4.1.87](https://netty.io/news/2023/01/12/4-1-87-Final.html)

Update versions in docs to be 2.0.58 instead of 2.0.58 for netty-tcnative-boringssl-static

### CODE REVIEW

**Comment by sdelamo** [comment]
@yawkat not sure if we should merge this into 3.8.x or into 3.9.x

**Comment by graemerocher** [comment]
3.8.x

File: http-server-netty/src/test/groovy/io/micronaut/http/server/netty/cors/NettyCorsSpec.groovy [hunk 1]
```diff
@@ -60,7 +60,7 @@ class NettyCorsSpec extends AbstractMicronautSpec {
         then:
         response.status == HttpStatus.NO_CONTENT
         headerNames.size() == 1
-        headerNames.contains("connection")
```
Comment (sdelamo) [inline http-server-netty/src/test/groovy/io/micronaut/http/server/netty/cors/NettyCorsSpec.groovy]: `contains` does not work in this `Set<String>`. It looks like a bug in netty.

**Review by yawkat — CHANGES_REQUESTED** [review]
pending https://github.com/netty/netty/pull/13219

**Comment by sdelamo** [comment]
Upgrading to [Netty 4.1.87](https://netty.io/news/2023/01/12/4-1-87-Final.html) which does not have the regression. 88 has a different issue and 89 has https://github.com/netty/netty/pull/13219

**Review by yawkat — APPROVED** [review]

**Pull request merged**